src/field_theory/intermediate_field.lean

```diff
@@ -280,10 +280,19 @@ set_like.coe_injective $ set.image_image _ _ _
 /-- Given an equivalence `e : L ≃ₐ[K] L'` of `K`-field extensions and an intermediate
 field `E` of `L/K`, `intermediate_field_equiv_map e E` is the induced equivalence
 between `E` and `E.map e` -/
-@[simps] def intermediate_field_map (e : L ≃ₐ[K] L') (E : intermediate_field K L) :
+def intermediate_field_map (e : L ≃ₐ[K] L') (E : intermediate_field K L) :
   E ≃ₐ[K] (E.map e.to_alg_hom) :=
 e.subalgebra_map E.to_subalgebra
 
+/- We manually add these two simp lemmas because `@[simps]` before `intermediate_field_map`
+  led to a timeout. -/
+@[simp] lemma intermediate_field_map_apply_coe (e : L ≃ₐ[K] L') (E : intermediate_field K L)
+  (a : E) : ↑(intermediate_field_map e E a) = e a := rfl
+
+@[simp] lemma intermediate_field_map_symm_apply_coe (e : L ≃ₐ[K] L') (E : intermediate_field K L)
+  (a : subsemiring.map e.to_ring_equiv.to_ring_hom E.to_subalgebra.to_subsemiring) :
```

Comment:
```suggestion
  (a : E.map e.to_alg_hom) :
```

Reply:
Both are defeq but not syntactically equal, also they don't dsimp to each other (your suggestion dsimp to `↥(E.map ↑e)` and the original dsimp to `↥(subsemiring.map ↑e.to_ring_hom E.to_subalgebra.to_subsemiring)`, so I think they won't fire in the same situations, but your suggestion is certainly much better in form. In general, should we aim to keep the auto-generated lemmas, or change them to a form that's more compact, like in your suggestion? Of course in this case either way is fine, because they're not used in mathlib yet.

Reply:
I do not quite understand how `simps` works, but in my experience it tends to over-"simplify" the LHS for the sake of simp normal form, to the point where you can't really recognize the original lemma. So I would be happy to keep the version that actually looks like the definition, aka Eric's.

Reply:
OK, I'll use Eric's version. So there are probably many `@[simps]`-generated lemma that could benefit from manual restatement.

Reply:
The point in my spelling is that it matches the type used to declare the map.

The problem with `simps` is not that it simplifies the LHS, but that it unfolds until it finds a constructor, and the constructor it finds appears to be too many levels down